src/Dapr.Extensions.Configuration/DaprConfigurationStoreProvider.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Dapr.Client;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Primitives;
+
+namespace Dapr.Extensions.Configuration
+{
+    /// <summary>
+    /// A configuration provider that utilizes the Dapr Configuration API. It can either be a single, constant
+    /// call or a streaming call.
+    /// </summary>
+    [Obsolete]
+    public class DaprConfigurationStoreProvider : ConfigurationProvider
+    {
+        private string store { get; }
+        private IReadOnlyList<string> keys { get; }
+        private DaprClient daprClient { get; }
+        private TimeSpan sidecarWaitTimeout { get; }
+        private bool isStreaming { get; } = false;
+        private IReadOnlyDictionary<string, string>? metadata { get; } = default;
+        private CancellationToken cancellationToken { get; } = default;
+        private Task subscribeTask = Task.CompletedTask;
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="store">The configuration store to query.</param>
+        /// <param name="keys">The keys, if any, to request. If empty, returns all configuration items.</param>
+        /// <param name="daprClient">The <see cref="DaprClient"/> used for the request.</param>
+        /// <param name="sidecarWaitTimeout">The <see cref="TimeSpan"/> used to configure the timeout waiting for Dapr.</param>
+        /// <param name="isStreaming">Determines if the source is streaming or not.</param>
+        /// <param name="metadata">Optional metadata sent to the configuration store.</param>
+        /// <param name="cancellationToken"><see cref="CancellationToken"/> used to cancel the operation.</param>
+        public DaprConfigurationStoreProvider(
+            string store,
+            IReadOnlyList<string> keys,
+            DaprClient daprClient,
+            TimeSpan sidecarWaitTimeout,
+            bool isStreaming = false,
+            IReadOnlyDictionary<string, string>? metadata = default,
+            CancellationToken cancellationToken = default)
```

Comment:
I think we should remove this `cancellationToken` parameter. This is a very tricky design and it will be hard for users to get the right cancellation behavior.

Instead we should implement `Dispose` and use that to cancel.